### PR TITLE
Fftw3 parallel support

### PR DESCRIPTION
**Pull request by dpshelio**
This allows use components like openmp or threads for fftw3.

Still needs to add support to combined components, eg: double and openmp together.

### CODE REVIEW

File: modules/FindFFTW3.cmake [hunk 1]
```diff
@@ -23,41 +23,155 @@ if(FFTW3_INCLUDE_DIR AND FFTW3_LIBRARIES)
...
-    set(FFTW3_FIND_COMPONENTS ${fftw3_FIND_COMPONENTS})
-endif()
+macro(find_specific_libraries KIND PARALLEL)
+  list(APPEND FFTW3_FIND_COMPONENTS ${KIND}_${PARALLEL})
+  message(STATUS "========= ${${PARALLEL}_FOUND} ===========")
+  if(NOT (${PARALLEL} STREQUAL "SERIAL") AND NOT ${PARALLEL}_FOUND)
+    message(FATAL_ERROR "Please, find ${PARALLEL} libraries before FFTW")
+  endif()
+
+  message(STATUS "Checking for ${KIND} and ${PARALLEL}")
```
Comment (mdavezac) [inline modules/FindFFTW3.cmake]: Try and use `<package>_FIND_QUIETLY` rather than  always output something.
We shouldn't leave debug message behind.

**Comment by dpshelio** [comment]
I think it's good to go now, only problems are:
 - full version support (without defaulting to `SERIAL`)
 - windows checks.

File: modules/FindFFTW3.cmake [hunk 2]
```diff
@@ -23,41 +26,142 @@ if(FFTW3_INCLUDE_DIR AND FFTW3_LIBRARIES)
...
+    # adding target properties to the different cases
+    ##   MPI
+    if(PARALLEL STREQUAL "MPI")
+      if(MPI_C_LIBRARIES)
+        set_target_properties(fftw3::${kind}::mpi PROPERTIES
+          IMPORTED_LOCATION "${FFTW3_${KIND}_${PARALLEL}_LIBRARY}"
+          INTERFACE_INCLUDE_DIRECTORIES "${FFTW3_INCLUDE_DIR_PARALLEL}"
+          IMPORTED_LINK_INTERFACE_LIBRARIES ${MPI_C_LIBRARIES})
+      endif()
+        # Below seems to be the same path than include_dir_parallel but it fails
```
Comment (mdavezac) [inline modules/FindFFTW3.cmake]: using version control means we shouldn't leave code that's commented out

Comment (dpshelio) [inline modules/FindFFTW3.cmake]: Oh yes! That's a bit that I didn't understood what was it for. I can remove it.

**Comment by mdavezac** [comment]
retest this please

**Pull request merged**